pygmt/base_plotting.py

```diff
@@ -853,42 +865,102 @@ def text(
-            a column.
+            justification is explicitly given (i.e. justify=True), then the
+            input textfile(s) must have this as a column.
         {J}
         {R}
+        clearance : str
+            ``[dx/dy][+to|O|c|C]``
+            Adjust the clearance between the text and the surrounding box
+            [15%]. Only used if *pen* or *fill* are specified. Append the unit
+            you want ('c' for cm, 'i' for inch, or 'p' for point; if not given
+            we consult 'PROJ_LENGTH_UNIT') or '%' for a percentage of the
+            font size. Optionally, use modifier '+t' to set the shape of the
+            textbox when using *fill* and/or *pen*. Append lower case 'o' to
+            get a straight rectangle [Default]. Append upper case 'O' to get a
+            rounded rectangle. In paragraph mode (*paragraph*) you can also
+            append lower case 'c' to get a concave rectangle or append upper
+            case 'C' to get a convex rectangle.
+        fill : str
+            Sets the shade or color used for filling the text box [Default is
+            no fill].
+            instead.
+        offset : str
+            ``[j|J]dx[/dy][+v[pen]]``
+            Offsets the text from the projected (x,y) point by dx,dy [0/0]. If
+            dy is not specified then it is set equal to dx. Use offset='j' to
+            offset the text away from the point instead (i.e., the text
+            justification will determine the direction of the shift). Using
+            offset='J' will shorten diagonal offsets at corners by sqrt(2).
+            Optionally, append '+v' which will draw a line from the original
+            point to the shifted point; append a pen to change the attributes
+            for this line.
+        pen : str
+            Sets the pen used to draw a rectangle around the text string
+            (see *clearance*) [Default is width = default, color = black,
+            style = solid].
         """
         kwargs = self._preprocess(**kwargs)
 
-        kind = data_kind(textfiles, x, y, text)
+        # Ensure inputs are either textfiles, x/y/text, or position/text
+        if position is None:
+            kind = data_kind(textfiles, x, y, text)
+        elif position is not None:
+            if x is not None or y is not None:
+                raise GMTInvalidInput(
+                    "Provide either position only, or x/y pairs, not both"
+                )
+            kind = "vectors"
+
         if kind == "vectors" and text is None:
-            raise GMTInvalidInput("Must provide text with x and y.")
+            raise GMTInvalidInput("Must provide text with x/y pairs or position")
         if kind == "file":
             for textfile in textfiles.split(" "):  # ensure that textfile(s) exist
                 if not os.path.exists(textfile):
                     raise GMTInvalidInput(f"Cannot find the file: {textfile}")
```

Comment:
We shouldn't check if text files exist since GMT not only search input files in the current directory but also search for files in the `~/.gmt`, `~/.gmt/cache` and `~/.gmt/server` directories.

Reply:
You're right, I forgot about the `@...` files. I feel like there was a reason I added this in commit 7a08ebd7a5d1b185f31b84265617d1816d2d8610, but can't recall now.

Reply:
Ok done, a missing file will now raise a GMTClibError instead of GMTInvalidInput as before:

```python
pygmt.exceptions.GMTCLibError: Module 'text' failed with status code 71:
text [ERROR]: Error for input file: No such file (notexist.txt)
```

so removing the `os.path.exists()` check should be fine.